src/transformers/commands/add_new_model_like.py

```diff
@@ -173,6 +173,63 @@ def parse_module_content(content: str) -> List[str]:
     return objects
 
 
+def extract_blocks(content: str, indent_level: int = 0) -> List[str]:
+    """Parse the blocks in `content` into the list of objects it defines.
+
+    The first line in `content` should be indented at `indent_level` level, otherwise an error will be thrown.
+
+    This method will immediately stop the search when a (non-empty) line with indent level less than `indent_level` is
+    encountered.
+
+    Args:
+        content (`str`): The content to parse
+        indent_level (`int`, *optional*, default to 0): The indent level of the blocks to search for
+
+    Returns:
+        `List[str]`: The list of objects defined in `content` with the indent level `indent_level`.
+    """
+    objects = []
+    current_object = []
+    lines = content.split("\n")
+    # Doc-styler takes everything between two triple quotes in docstrings, so we need a fake """ here to go with this.
+    end_markers = [")", "]", "}", '"""']
+
+    for idx, line in enumerate(lines):
+        if idx == 0 and indent_level > 0 and not is_empty_line(line) and find_indent(line) != indent_level:
+            raise ValueError(
+                f"When `indent_level > 0`, the first line in `content` should have indent level {indent_level}. Got "
+                f"{find_indent(line)} instead."
+            )
+
+        if find_indent(line) < indent_level and not is_empty_line(line):
+            break
+
+        # End of an object
+        is_valid_object = len(current_object) > 0
+        if (
+            not is_empty_line(line)
+            and not line.endswith(":")
```

Comment:
This check is necessary. Othewise

```python
    def is_pipeline_test_to_skip(
        self, pipeline_test_casse_name, config_class, model_architecture, tokenizer_name, processor_name
    ):
        return False
```
will enter the `if block` here and closing the `current_object` at the 3rd line `):`.

I believe this should be added to `parse_module_content` too, but it's super rare to have such case at indent level 0. And I don't want to make any change to that method as you pointed before.

Reply:
Fine for me to add it in the other method as well. But it can be in another PR if you want.